vms/platformvm/state/state_test.go

```diff
@@ -88,8 +91,604 @@ func TestStateSyncGenesis(t *testing.T) {
+				r.ErrorIs(err, database.ErrNotFound)
+
+				_, err = s.flatValidatorPublicKeyDiffsDB.Get(marshalDiffKey(staker.SubnetID, height, staker.NodeID))
+				r.ErrorIs(err, database.ErrNotFound)
+			},
+			reloadState: func(r *require.Assertions, rebuiltState *state) {
+				r.NoError(rebuiltState.loadPendingValidators())
+				r.NoError(rebuiltState.initValidatorSets())
+			},
+		},
+		"delete pending delegator": {
+			storeStaker: func(r *require.Assertions, subnetID ids.ID, s *state) *Staker {
+				// insert validator and delegator the remove the validator
+				var (
+					valStartTime = time.Now().Truncate(time.Second).Unix()
+					delStartTime = time.Unix(valStartTime, 0).Add(time.Hour).Unix()
+					delEndTime   = time.Unix(delStartTime, 0).Add(30 * 24 * time.Hour).Unix()
+					valEndTime   = time.Unix(valStartTime, 0).Add(365 * 24 * time.Hour).Unix()
+
+					validatorsData = txs.Validator{
+						NodeID: ids.GenerateTestNodeID(),
+						Start:  uint64(valStartTime),
+						End:    uint64(valEndTime),
+						Wght:   1234,
+					}
+
+					delegatorData = txs.Validator{
+						NodeID: validatorsData.NodeID,
+						Start:  uint64(delStartTime),
+						End:    uint64(delEndTime),
+						Wght:   validatorsData.Wght / 2,
+					}
+				)
+
+				utxVal := createPermissionlessValidatorTx(r, subnetID, validatorsData)
+				addPermValTx := &txs.Tx{Unsigned: utxVal}
+				r.NoError(addPermValTx.Initialize(txs.Codec))
+
+				val, err := NewPendingStaker(addPermValTx.ID(), utxVal)
+				r.NoError(err)
+
+				utxDel := createPermissionlessDelegatorTx(subnetID, delegatorData)
+				addPermDelTx := &txs.Tx{Unsigned: utxDel}
+				r.NoError(addPermDelTx.Initialize(txs.Codec))
+
+				del, err := NewPendingStaker(addPermDelTx.ID(), utxDel)
+				r.NoError(err)
+
+				s.PutPendingValidator(val)
+				s.AddTx(addPermValTx, status.Committed) // this is currently needed to reload the staker
+
+				s.PutPendingDelegator(del)
+				s.AddTx(addPermDelTx, status.Committed) // this is currently needed to reload the staker
+				r.NoError(s.Commit())
+
+				s.DeletePendingDelegator(del)
+				r.NoError(s.Commit())
+				return del
+			},
+			checkStoredStakerData: func(r *require.Assertions, s *state, staker *Staker, height uint64) {
+				// Check state validator is stored in P-chain state
+				delIt, err := s.GetPendingDelegatorIterator(staker.SubnetID, staker.NodeID)
+				r.NoError(err)
+				r.False(delIt.Next())
+				delIt.Release()
+
+				// Check that validator is made available in the validators set, with the right weight
+				valsMap := s.cfg.Validators.GetMap(staker.SubnetID)
+				r.Len(valsMap, 0)
+			},
+			reloadState: func(r *require.Assertions, rebuiltState *state) {
+				r.NoError(rebuiltState.loadPendingValidators())
+				r.NoError(rebuiltState.initValidatorSets())
+			},
+		},
+	}
+
+	subnetIDs := []ids.ID{constants.PrimaryNetworkID, ids.GenerateTestID()}
+	for _, subnetID := range subnetIDs {
+		for name, test := range tests {
+			t.Run(name, func(t *testing.T) {
```

Comment:
We should probably include the subnet ID in the test name too, right? Otherwise we'll be running two different tests with the same name, which could make debugging confusing

Reply:
Done